### ISSUE TITLE
A disabled plugin is automatically started by pluginManager.startPlugins()

### ISSUE THREAD

**Issue by hptvl** [pf4j/pf4j#223]
Regarding to plugin life-cycle description in pf4j homepage (https://pf4j.org/doc/plugin-lifecycle.html),

> "The DefaultPluginManager contains the following logic:
> DISABLED plugins are NOT automatically STARTED by pf4j in startPlugins() BUT you may manually start (and therefore enable) a DISABLED plugin by calling startPlugin(pluginId) instead of enablePlugin(pluginId) + startPlugin(pluginId)"

In my main application, there is a plugin and a disabled.txt file which contain plugin id needs to be disabled in plugins folder. However, when DefaultPluginManager.startPlugins() is executed, the disabled plugin could be able to be started.
 
```java
PluginManager pluginManager = new DefaultPluginManager();
pluginManager.loadPlugins();
List<PluginWrapper> plugins = pluginManager.getPlugins();
pluginManager.startPlugins();
for (PluginWrapper w : plugins) {
    System.out.println(w.getPluginId() + "----" + w.getPluginState().toString());
}
```
I thought the problem is AbstractPluginManager.resolvePlugins() probably changes plugin state from "UNRESOLVED" (even its state is DISABLED) to "RESOLVED". Once, the disabled plugin's state is changed to RESOLVED, it could be started by startPlugins().

**Comment by decebals** [comment]
I think that I resolved it. This fix comes with a test.
This fix comes with code from #204 PR, thanks @hxzhao527!

**Comment by hxzhao527** [comment]
:blush It's my pleasure.
The PR now is out of date, I will close it.
